### PR TITLE
Have consistent compile lines between BUILD_TESTS enabled or not

### PR DESCRIPTION
**Pull request by robertmaynard**
This will remove 1h from our conda CI builds since we can now re-use the cached object files between `libraft` and `libraft-tests`

### CODE REVIEW

**Review by vyasr — APPROVED** [review]
Awesome

**Review by benfred — APPROVED** [review]

**Review by divyegala — APPROVED** [review]

**Comment by robertmaynard** [comment]
Do we have a workaround/PR for the `dask-cuda==23.6.0a0+10.g80596bd` and `distributed 2023.3.2+29.g1ccc312a` conflict I am getting in the wheel builds?

**Comment by divyegala** [comment]
> Do we have a workaround/PR for the dask-cuda==23.6.0a0+10.g80596bd and distributed 2023.3.2+29.g1ccc312a conflict I am getting in the wheel builds?

@galipremsagar is this similar to what is happening in other repos?

**Comment by galipremsagar** [comment]
> > Do we have a workaround/PR for the dask-cuda==23.6.0a0+10.g80596bd and distributed 2023.3.2+29.g1ccc312a conflict I am getting in the wheel builds?
> 
> @galipremsagar is this similar to what is happening in other repos?

Yes, `23.06` is blocked until this is merged: https://github.com/rapidsai/raft/pull/1403

**Comment by galipremsagar** [comment]
CI should now be unblocked. @robertmaynard you can merge the `branch-23.06` changes into your PR.

**Comment by robertmaynard** [comment]
> UPDATE: CI should now be unblocked. @robertmaynard you can merge the `branch-23.06` changes into your PR.

Thanks!

**Pull request merged**
